### ISSUE TITLE
Changing the deployment target of the project created by pod lib create

### ISSUE THREAD

**Issue by Isuru-Nanayakkara** [CocoaPods/CocoaPods#9900]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

This is not an issue with CocoaPods. This is a personal question. I did ask this on Stack Overflow but did not receive a sufficient answer so I'm re-posting the question here. Sorry about that.

---

I created a CocoaPod library project for iOS via the `pod lib create` command.

The `podspec` file that gets created adds the deployment target as iOS 8 in it. I want it to be iOS 11. There is no option to set it at the beginning as far as I know.

If I [change it manually](https://i.imgur.com/g71eBoa.png) in the Xcode project and run `pod install` again, it creates a static framework instead of a dynamic one.
How do I set/change the deployment target to something other than the default value? Is there a way to like "regenerate" the Xcode project to make the pod take this new deployment target into effect?

**Comment by dnkoutso** [comment]
I dont fully understand the question. I believe iOS 9 was when dynamic frameworks were introduced. I need to see a sample project with a podfile to be able to fully answer your question.

`use_frameworks!` or not in the `Podfile` tells CocoaPods whether to use frameworks for all pod targets or not.

**Comment by Isuru-Nanayakkara** [comment]
Hi, thanks for the response. I think I made it confusing it by talking about stati/dynamic frameworks. Let me rephrase my question.

I want to create a CocoPod library which supports from iOS 11 upwards. I ran the `pod lib create` command with the following options.

```
isurunanayakkara@isurus-mac-mini Desktop % pod lib create MyLibrary
Cloning `https://github.com/CocoaPods/pod-template.git` into `MyLibrary`.
Configuring MyLibrary template.

------------------------------

To get you started we need to ask a few questions, this should only take a minute.

If this is your first time we recommend running through with the guide: 
 - https://guides.cocoapods.org/making/using-pod-lib-create.html
 ( hold cmd and double click links to open in a browser. )


What platform do you want to use?? [ iOS / macOS ]
 > iOS

What language do you want to use?? [ Swift / ObjC ]
 > Swift

Would you like to include a demo application with your library? [ Yes / No ]
 > Yes

Which testing frameworks will you use? [ Quick / None ]
 > None

Would you like to do view based testing? [ Yes / No ]
 > No

Running pod install on your new library.

Analyzing dependencies
Downloading dependencies
Installing MyLibrary (0.1.0)
Generating Pods project
Integrating client project

[!] Please close any current Xcode sessions and use `MyLibrary.xcworkspace` for this project from now on.
Pod installation complete! There is 1 dependency from the Podfile and 1 total pod installed.

[!] Automatically assigning platform `iOS` with version `9.3` on target `MyLibrary_Example` because no platform was specified. Please specify a platform for this target in your Podfile. See `https://guides.cocoapods.org/syntax/podfile.html#platform`.

 Ace! you're ready to go!
 We will start you off by opening your project in Xcode
  open 'MyLibrary/Example/MyLibrary.xcworkspace'

To learn more about the template see `https://github.com/CocoaPods/pod-template.git`.
To learn more about creating a new pod, see `https://guides.cocoapods.org/making/making-a-cocoapod`.
```

Here is the [Xcode project](https://www.dropbox.com/s/la4pphvyq3xu0te/mylibrary.zip?dl=0) it creates automatically.

If you open it and click on the library pod, it shows the deployment target as iOS 8. 

![Screenshot 2020-07-05 at 09 03 35](https://user-images.githubusercontent.com/2551877/86525026-2f3fa880-be9f-11ea-9e35-ceed92d28503.png)

I want the deployment target to be iOS 11. Because my code has some API calls that are available from iOS 11.

So my question is how do I change my library's deployment target to iOS 11. Do I just change it manually here within Xcode and then in the pod spec file? Or is there another way to do this?

**Comment by dnkoutso** [comment]
Did you set `s.deployment_target` in your podspec?

**Comment by Isuru-Nanayakkara** [comment]
The podspec file gets created automatically when I run the `pod lib create` command. So I don't get to set anything in it until after it's created.

**Comment by dnkoutso** [comment]
Yes can you not just edit it?

**Comment by amorde** [comment]
You can edit it but sounds like a potential improvement to `pod lib create` to ask the user what deployment target version to use, and potentially update our current "minimum" for newly created podspecs from that command

**Comment by Isuru-Nanayakkara** [comment]
> Yes can you not just edit it?

Yes. But how should I make the pod that's added to the Xcode projects change its deployment target after its created? Do I manually change it to iOS 11. Is that the proper way to do it?

**Comment by Isuru-Nanayakkara** [comment]
> You can edit it but sounds like a potential improvement to `pod lib create` to ask the user what deployment target version to use, and potentially update our current "minimum" for newly created podspecs from that command

Yes, this would be a really good addition.

**Comment by amorde** [comment]
After creating the podspec you can edit it and then run `pod install`

**Comment by Isuru-Nanayakkara** [comment]
Thanks guys. I got it working. Appreciate both your help.
